pandas/tseries/period.py

```diff
@@ -271,9 +271,10 @@ def _from_arraylike(cls, data, freq, tz):
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
         if not getattr(values, 'dtype', None):
-            values = np.array(values, copy=False)
-        if is_object_dtype(values):
-            return PeriodIndex(values, name=name, freq=freq, **kwargs)
+            try:  # assume values / ints
+                values = np.array(values, copy=False, dtype='int64')
+            except TypeError:  # objects / Periods
```

Comment:
This is ugly and I would love a better way of doing it. 
I do think the outcomes are correct though - @sinhrks do you disagree?
Open to nice implementation though